[PowerPoint] (overview) Refresh (#4858)
OfficeDev/office-js-docs-pr: docs/powerpoint/powerpoint-add-ins.md

If you're building a content add-in, you'll need to get the presentation's active view and handle the `ActiveViewChanged` event, as part of your `Office.initialize` handler.

> [!NOTE]
> In PowerPoint on the web, the [Document.ActiveViewChanged](/javascript/api/office/office.document) event will never fire as Slide Show mode is treated as a new session. In this case, the add-in must fetch the active view on load, as shown in the following code sample.

Note the following about the code sample:

- The  `getActiveFileView` function calls the [Document.getActiveViewAsync](/javascript/api/office/office.document#office-office-document-getactiveviewasync-member(1)) method to return whether the presentation's current view is "edit" (any of the views in which you can edit slides, such as **Normal** or **Outline View**) or "read" (**Slide Show** or **Reading View**).

- The  `registerActiveViewChanged` function calls the [addHandlerAsync](/javascript/api/office/office.document#office-office-document-addhandlerasync-member(1)) method to register a handler for the [Document.ActiveViewChanged](/javascript/api/office/office.document) event.

```js

> // General Office.initialize function. Fires on loading the add-in.

Reviewer: ```suggestion
// General Office.initialize function. Called after the add-in loads and Office JS is initialized.
```

> Office.initialize = function(){

Reviewer: Should use Office.onReady()...

    Office.context.document.getActiveViewAsync(function (asyncResult) {

>         if (asyncResult.status == "failed") {

Reviewer: ```suggestion
        if (asyncResult.status === "failed") {
```

        function (asyncResult) {

>             if (asyncResult.status == "failed") {

Reviewer: ```suggestion
            if (asyncResult.status === "failed") {
```

## Navigate to a particular slide in the presentation

In the following code sample, the `getSelectedRange` function calls the [Document.getSelectedDataAsync](/javascript/api/office/office.document#office-office-document-getselecteddataasync-member(1)) method to get the JSON object returned by `asyncResult.value`, which contains an array named `slides`. The `slides` array contains the IDs, titles, and indexes of selected range of slides (or of the current slide, if multiple slides aren't selected). It also saves the ID of the first slide in the selected range to a global variable.

```js
function getSelectedRange() {
    // Gets the ID, title, and index of the current slide (or selected slides) and store the first slide ID. */
    Globals.firstSlideId = 0;

    Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, function (asyncResult) {

>         if (asyncResult.status == "failed") {

Reviewer: ```suggestion
        if (asyncResult.status === "failed") {
```

    Office.context.document.goToByIdAsync(Globals.firstSlideId, Office.GoToType.Slide, function (asyncResult) {

>         if (asyncResult.status == "failed") {

Reviewer: ```suggestion
        if (asyncResult.status === "failed") {
```

    Office.context.document.goToByIdAsync(goToNext, Office.GoToType.Index, function (asyncResult) {

>         if (asyncResult.status == "failed") {

Reviewer: ```suggestion
        if (asyncResult.status === "failed") {
```

```js
function getFileUrl() {
    // Gets the URL of the current file.
    Office.context.document.getFilePropertiesAsync(function (asyncResult) {
        const fileUrl = asyncResult.value.url;

>         if (fileUrl == "") {

Reviewer: ```suggestion
        if (fileUrl === "") {
```
